server/api/api/utils.py

```diff
@@ -223,22 +223,58 @@ def mask_notification_params_on_task(
     action: server.api.constants.MaskOperations,
 ):
     """
-    Mask notification config params from the task object
+    Mask notification config params from the task dictionary
     :param task:    The task object to mask
     :param action:  The masking operation to perform on the notification config params (conceal/redact)
     """
-    mask_op = _notification_params_mask_op(action)
     run_uid = get_in(task, "metadata.uid")
     project = get_in(task, "metadata.project")
     notifications = task.get("spec", {}).get("notifications", [])
-    masked_notifications = []
     if notifications:
-        for notification in notifications:
-            notification_object = mlrun.model.Notification.from_dict(notification)
-            masked_notifications.append(
-                mask_op(project, run_uid, notification_object).to_dict()
-            )
-        task.setdefault("spec", {})["notifications"] = masked_notifications
+        task.setdefault("spec", {})["notifications"] = _mask_notifications_params(
```

Comment:
Looks like in this flow you're missing a `to_dict()` on the masked notifications - before your change the code would do `mask_op(...).to_dict()`, which is something you don't do now.